mantidproject/mantid
# Keep up with nexus for D17

**Description of work.**
This introduces explicit time dependence for some critical hard-coded parameters for D17.
They are not set in the nexus files, from the other hand time-dependent instrument parameters one cannot have without duplicating the IDF. So this is probably the best compromise in interim. Proper solution will be set forth in https://github.com/mantidproject/mantid/issues/29505
Also adjusts some default parameters as requested by the instrument scientist.

**To test:**
Existing tests should pass
Load the newly added unit test file: `ILL/D17/564343.nxs`.
Check the sample logs, make sure `ChopperGap=0.0752 m` and `ChopperWindow=20 degree`.



*There is no associated issue.*





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
